Implement API level tests

BarisSari opened the pull request:
Test are moved from employer-callback repo.

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+@mock.patch("app.use_cases.create_enrolment.CreateNewEnrolment")
+def test_create_enrolment_success(use_case):
+    message = "The callback has been fetched from the server."
```
enrolment*

BarisSari replied:
corrected

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+
+@mock.patch("app.use_cases.create_enrolment.CreateNewEnrolment")
+def test_create_enrolment_success(use_case):
```
there is no method in the code to fetch enrolment

BarisSari replied:
test deleted

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+@mock.patch("app.use_cases.get_enrolment.GetEnrolmentByID")
+def test_get_enrolment_by_id_success_created(use_case):
+    code = SuccessType.CREATED
+    message = "The enrolment has been created."
```
"Enrolment has been fetched."

BarisSari replied:
deleted the test

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+
+@mock.patch("app.use_cases.get_enrolment.GetEnrolmentByID")
+def test_get_enrolment_by_id_success_created(use_case):
```
GetEnrolmentByID doesnt create anything

BarisSari replied:
removed

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+@mock.patch("app.use_cases.get_enrolment.GetEnrolmentByID")
+def test_get_enrolment_by_id_success(use_case):
+    message = "The callback has been fetched from the server."
```
enrolment*

BarisSari replied:
corrected

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+
+@mock.patch("app.use_cases.get_enrolment.GetEnrolmentByID")
+def test_get_enrolment_by_id_success(use_case):
```
i think only this test method needed. not previous for created

BarisSari replied:
created method deleted

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+
+@mock.patch("app.use_cases.get_enrolment_status.GetEnrolmentStatus")
+def test_get_enrolment_status_success_created(use_case):
```
GetEnrolmentStatus doesnt create anything

BarisSari replied:
deleted

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+@mock.patch("app.use_cases.get_enrolment_status.GetEnrolmentStatus")
+def test_get_enrolment_status_success_created(use_case):
+    code = SuccessType.CREATED
+    message = "The enrolment has been created."
```
"Enrolment status has been fetched."

BarisSari replied:
I removed it because it won't create anything. This is a GET method

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+@mock.patch("app.use_cases.get_enrolment_status.GetEnrolmentStatus")
+def test_get_enrolment_status_success(use_case):
+    message = "The callback has been fetched from the server."
```
"Enrolment status has been fetched."

BarisSari replied:
updated

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+@mock.patch("app.use_cases.get_callbacks_list.GetCallbacksList")
+def test_get_callbacks_list_for_enrolment_success_created(use_case):
+    code = SuccessType.CREATED
+    message = "The enrolment has been created."
```
"Callbacks list has been fetched"

BarisSari replied:
removed

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+@mock.patch("app.use_cases.get_callbacks_list.GetCallbacksList")
+def test_get_callbacks_list_for_enrolment_success(use_case):
+    message = "The callback has been fetched from the server."
```
"Callbacks list has been fetched"

BarisSari replied:
updated

TheAyeshaSiddiqa commented on tests/api/test_enrolments.py:
```diff
@@ -0,0 +1,256 @@
+
+
+@mock.patch("app.use_cases.get_callbacks_list.GetCallbacksList")
+def test_get_callbacks_list_for_enrolment_success_created(use_case):
```
GetCallbacksList doesnt create anything

BarisSari replied:
removed

TheAyeshaSiddiqa reviewed (changes requested):
I think you moved the code without changes.

BarisSari commented:
> I think you moved the code without changes.

I thought I moved the code after the changes but you're right. Sorry, for the double-review. I corrected everything.
